modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+
+  params.addRangeCheckedParam<std::vector<unsigned int>>(
+      "mesh_intervals",
+      std::vector<unsigned int>{1},
+      "mesh_intervals>0",
+      "The number of meshing intervals for each region starting at the center (length(ring_radii) "
+      "+ "
+      "length(duct_halfpitch )+ 1)");
+
+  params.addParam<std::vector<std::vector<subdomain_id_type>>>(
+      "region_ids",
+      "IDs for each radial and axial zone for assignment of unique physical characteristics");
+
+  params.addParam<bool>("extrude",
+                        false,
+                        "Determines if this is the final step in the geometry construction"
+                        " and extrudes the 2D geometry to 3D. If this is true then this mesh "
+                        "cannot be used in further mesh building");
+
+  params.addParam<bool>(
+      "quad_center_elements", true, "Whether the center elements are quad or triangular.");
+
+  params.addClassDescription("This PinMeshGenerator object is designed to generate pin-like "
+                             "structures, with IDs, from a reactor geometry. "
+                             "Whether it be a square or hexagonal pin, they are divided into three "
+                             "substructures - the innermost "
+                             "radial pin regions, the single bridging background region, and the "
+                             "square or hexagonal ducts regions.");
+
+  return params;
+}
+
+PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _reactor_params(getParam<MeshGeneratorName>("reactor_params")),
+    _pin_type(getParam<subdomain_id_type>("pin_type")),
+    _pitch(getParam<Real>("pitch")),
+    _num_sectors(getParam<unsigned int>("num_sectors")),
+    _ring_radii(isParamValid("ring_radii") ? getParam<std::vector<Real>>("ring_radii")
+                                           : std::vector<Real>()),
+    _duct_halfpitch(isParamValid("duct_halfpitch") ? getParam<std::vector<Real>>("duct_halfpitch")
+                                                   : std::vector<Real>()),
+    _intervals(getParam<std::vector<unsigned int>>("mesh_intervals")),
+    _region_ids(isParamValid("region_ids")
+                    ? getParam<std::vector<std::vector<subdomain_id_type>>>("region_ids")
+                    : std::vector<std::vector<subdomain_id_type>>()),
+    _extrude(getParam<bool>("extrude")),
+    _quad_center(getParam<bool>("quad_center_elements"))
+{
+  declareMeshProperty("pitch", _pitch);
+
+  // Ensure that the user has supplied the correct info for conformal mesh generation
+  if (getMeshByName(_reactor_params) != nullptr)
+    mooseError("The reactor_params mesh is not of the correct type");
+
+  if (!hasMeshProperty("mesh_dimensions", _reactor_params) ||
+      !hasMeshProperty("mesh_geometry", _reactor_params))
+    mooseError("The reactor_params input must be a ReactorMeshParams type MeshGenerator\n");
```

Comment:
this gets triggered RMP is physically below PMG in the input file.
We need dependency resolution (see how input is being handled for mesh generators)

Reply:
Isn't there dependency resolution already built in? PMG requires "reactor_params" as an input, so there's no way the input file would allow RMG to be defined below PMG

Reply:
ah you're right. The error message is just clearer for those though. 
Maybe add a line to this error (and same in Assembly / Core) to say RMP should be defined before

Reply:
Added more detail to the error message